### ISSUE TITLE
Mac web_tool_tests has become very flaky

### ISSUE THREAD

**Issue by zanderso** [flutter/flutter#123018]
Failures are in hot restart tests. In the past 6 flakes, there have been failures in two different unit tests. See:

https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20web_tool_tests/11270/overview and
https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20web_tool_tests/11260/overview

The failures look similar though. From the first link:

```
[  +53 ms] DwdsVmClient: Successful hot restart
[   +2 ms] ChromeProxyService: Caching package:flutter/src/widgets/widget_inspector.dart in expression compiler worker
[        ] DevHandler: VmService proxy responded with an error:
{jsonrpc: 2.0, id: 15, error: {code: -32603, message: getIsolate: [Sentinel kind: Collected, valueAsString: Unrecognized isolateId: 1] from getIsolate(), data: {details: #0      ChromeProxyService._throwSentinel (package:dwds/src/services/chrome_proxy_service.dart:1154:5)
#1      ChromeProxyService._checkIsolate (package:dwds/src/services/chrome_proxy_service.dart:1146:7)
#2      ChromeProxyService.getIsolate.<anonymous closure> (package:dwds/src/services/chrome_proxy_service.dart:569:7)
<asynchronous suspension>
#3      StackZoneSpecification._registerUnaryCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:124:15)
<asynchronous suspension>
}}}
```

From the second link:
```
[  +49 ms] DwdsVmClient: Successful hot restart
[   +1 ms] ChromeProxyService: Caching package:flutter/src/widgets/widget_inspector.dart in expression compiler worker
[        ] DevHandler: VmService proxy responded with an error:
{jsonrpc: 2.0, id: 15, error: {code: -32603, message: getIsolate: [Sentinel kind: Collected, valueAsString: Unrecognized isolateId: 1] from getIsolate(), data: {details: #0      ChromeProxyService._throwSentinel (package:dwds/src/services/chrome_proxy_service.dart:1154:5)
#1      ChromeProxyService._checkIsolate (package:dwds/src/services/chrome_proxy_service.dart:1146:7)
#2      ChromeProxyService.getIsolate.<anonymous closure> (package:dwds/src/services/chrome_proxy_service.dart:569:7)
<asynchronous suspension>
#3      StackZoneSpecification._registerUnaryCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:124:15)
<asynchronous suspension>
}}}
```

**Comment by zanderso** [comment]
@elliette for dwds.

**Comment by zanderso** [comment]
I see @christopherfujino also recently touched this file in https://github.com/flutter/flutter/pull/122855

**Comment by zanderso** [comment]
And @annagrin in https://github.com/flutter/flutter/pull/122776

**Comment by annagrin** [comment]
I fixed the flake in https://github.com/flutter/flutter/pull/122776 but it got reverted, I suspect due to new tests I've added that took the whole file over some threshold. @yjbanov @zanderso @christopherfujino the fixes are currently all in flutter tools. Would you like to take over this? Alternatively I could re-submit my fix and split the tests in two files. Hot restart tests are quite important so we don't want to keep them disabled for very long.

**Comment by christopherfujino** [comment]
> I fixed the flake in #122776 but it got reverted, I suspect due to new tests I've added that took the whole file over some threshold. @yjbanov @zanderso @christopherfujino the fixes are currently all in flutter tools. Would you like to take over this? Alternatively I could re-submit my fix and split the tests in two files. Hot restart tests are quite important so we don't want to keep them disabled for very long.

I'll attempt a re-land with more shards.

**Comment by annagrin** [comment]
@christopherfujino FYI

@derekxu16 is currently working on a proper fix that will work for all vmService API so another option is to wait for the fix in the SDK to propagate (my patch should not be needed in that case): https://github.com/dart-lang/sdk/issues/51752

**Comment by derekxu16** [comment]
Just published `package:vm_service v11.2.1`. The problem should be fixed once that gets rolled into the Flutter SDK.

https://pub.dev/packages/vm_service/versions/11.2.1
